src/display.rs

```diff
@@ -106,21 +106,17 @@ pub trait DisplayHex: Copy + sealed::IsRef {
     fn hex_reserve_suggestion(self) -> usize { 0 }
 }
 
-fn internal_display(bytes: &[u8], f: &mut fmt::Formatter, case: Case) -> fmt::Result {
-    use fmt::Write;
-    // There are at least two optimizations left:
-    //
-    // * Reusing the buffer (encoder) which may decrease the number of virtual calls
-    // * Not recursing, avoiding another 1024B allocation and zeroing
-    //
-    // This would complicate the code so I was too lazy to do them but feel free to send a PR!
```

Comment:
I rekon these comment should be written up into an issue (or 2) and removed - assuming you understand them. (I would have to spend some time thinking to work them out.)

Reply:
In the commits below I keep these comments around - if that's ok with you we can address these in a follow-up PR.

Reply:
Sure thing.